packages/data-stores/src/site/actions.ts

```diff
@@ -57,18 +63,43 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 		}
 	}
 
+	const receiveExistingSite = ( slug: string, response: ExistingSiteDetails | undefined ) => ( {
+		type: 'RECEIVE_EXISTING_SITE' as const,
+		slug,
+		response,
+	} );
+
+	const receiveExistingSiteFailed = ( slug: string, response: ExistingSiteError | undefined ) => ( {
```

Comment:
On a sidenote, dedicated actions (and normally, reducers) for failed network requests _is_ a pattern that we have in Calypso, but I'm also not totally sure I like it. It means a lot of boilerplate (adding such actions and reducers for each request -- somewhat similar to [`isFetching...` state](https://github.com/Automattic/wp-calypso/pull/40158/files#r398577848)), and I find that we're not even always using that error state in the UI.

OTOH, I've rarely seen this pattern in Gutenberg (IIRC). 

Definitely not something we need to address as part of this PR, but I think it's worth discussing how to handle this sort of errors in Gutenboarding's (and Calypso's) future. Maybe we should just `throw` from the failing network handler, and use [React Error Boundaries](https://reactjs.org/docs/error-boundaries.html). Not sure if there's an established pattern nowadays.

/cc @Automattic/team-calypso @youknowriad @aduth @sirreal

Reply:
We do have network request errors in state I believe, the difference is that we have the "entities" abstractions, so we don't recreate these actions/selectors/reducer per API, it's all handled in the same one.

Reply:
That's a good point about errors. In this case we're not (yet) displaying the error, but the pattern was included to match the `user` data store where we are using the errors on the front end, and performing another action is designed to clear the error state. That doesn't necessarily need to live in the reducer though, and we could instead expect the component UI to worry about how long to keep around the message itself, so I think you're right, this could be a good area to explore simplifying what we're keeping in the data store in the future.